web/src/clipboard_handler.ts

```diff
@@ -0,0 +1,42 @@
+import * as hash_util from "./hash_util";
+import * as stream_data from "./stream_data";
+
+export function generate_formatted_link(link_text: string): string {
+    const stream_topic = hash_util.decode_stream_topic_from_url(link_text);
```

Comment:
This should probably be called `link_data`, since a `message_id` can be in that object too.